### PR TITLE
bazel: Bugfix, Update to version 6.1.2; PG bazel

### PR DESCRIPTION
**Pull request by essandess**
#### Description

Cherry picked from #15397.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6 21G115 x86_64
Xcode 14.0 14A309

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by essandess** [comment]
CI error:
```
2022-09-17T21:05:01.2760470Z [31m[1mFATAL: [0mbazel crashed due to an internal error. 
```

I do not observe this with a trace build. I recommend that a reviewer confirm this and merge whether or not the CI completed.

`bazel` is unreliable in the best of times, and at least this PR will give us a working up-to-date `bazel` binary. Currently, the ` bazel ` port is old **and** broken.

**Comment by essandess** [comment]
I confirm that this latest version builds locally on:
macOS 12.6.1 21G217 x86_64
Xcode 14.1 14B47b

File: devel/bazel/Portfile [hunk 1]
```diff
@@ -410,4 +363,7 @@ post-destroot {
 
 notes "
 See http://bazel.build/docs/getting-started.html to start a new project!
+
+MacPorts bazel uses the compilers `cxx_bazelwrap` and `cc_bazelwrap`; set
```
[the remark this answers is not in the document]
Comment (essandess) [reply devel/bazel/Portfile]: Added, thank you.

File: devel/bazel/Portfile [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-PortSystem                                            1.0
-PortGroup             github                          1.0
-PortGroup             java                            1.0
-PortGroup             compiler_blacklist_versions     1.0
-PortGroup             legacysupport                   1.1
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           java 1.0
+PortGroup           compiler_blacklist_versions 1.0
+PortGroup           legacysupport 1.1
```
Comment (ryandesign) [inline devel/bazel/Portfile:7]: Lots of unrelated whitespace changes in this PR, making the functional changes more difficult to identify and review.

File: _resources/port1.0/group/bazel-1.0.tcl [hunk 3]
```diff
@@ -256,16 +274,15 @@ pre-build {
...
     }
     if { [option bazel.run_bazel_fetch] && [option bazel.build_cmd] ne "" && [file exists ${worksrcpath}] } {
-        set cc  [compwrap::wrap_compiler cc]
-        set cxx [compwrap::wrap_compiler cxx]
         # Run fetch
         set addpath [string map {" " ":"} [option bazel.path]]
         system -W ${worksrcpath} "PATH=${addpath}:$env(PATH) [bazel::get_build_env] [option bazel.build_cmd] [option bazel.build_cmd_opts] fetch [option bazel.build_target]"
         # Patch the bazel clang wrapper script to use MacPorts compiler selection and support ccache
         foreach f [ exec find [bazel::get_bazel_build_area] -name "wrapped_clang.cc" ] {
+            ui_msg "Patching '${f}'."
```
Comment (ryandesign) [inline _resources/port1.0/group/bazel-1.0.tcl:282]: Do we really want this additional line of output for each file being patched? No other port or portgroup lists files as they're being patched.

File: devel/bazel/Portfile [hunk 4]
```diff
@@ -29,115 +29,66 @@ set extra_args ""
...
-                      size    313677756
-
-    compiler.blacklist-append {clang < 710}
-    compiler.blacklist-append {macports-clang-[7-9].0}
-    set min_darwin    16
+    github.setup    bazelbuild ${name} 3.7.2
+    revision        1
 
+    # Obsolete Date: 2023-08-01
+    replaced_by     bazel
```
Comment (ryandesign) [inline devel/bazel/Portfile:55]: `replaced_by` doesn't work unless you increase `revision` or `version` or `epoch`.

Obsoleting these ports will break all the ports that depend on them, for example py-dm-tree.

**Review by ryandesign — COMMENTED** [review]
CI failed:
```
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.ExceptionInInitializerError
	at com.google.devtools.build.lib.actions.ParameterFile.writeContentLatin1(ParameterFile.java:141)
	at com.google.devtools.build.lib.actions.ParameterFile.writeContent(ParameterFile.java:120)
	at com.google.devtools.build.lib.actions.ParameterFile.writeParameterFile(ParameterFile.java:112)
	at com.google.devtools.build.lib.analysis.actions.ParameterFileWriteAction$ParamFileWriter.writeOutputFile(ParameterFileWriteAction.java:172)
	at com.google.devtools.build.lib.exec.FileWriteStrategy.beginWriteOutputToFile(FileWriteStrategy.java:58)
	at com.google.devtools.build.lib.analysis.actions.FileWriteActionContext.beginWriteOutputToFile(FileWriteActionContext.java:49)
	at com.google.devtools.build.lib.analysis.actions.AbstractFileWriteAction.beginExecution(AbstractFileWriteAction.java:66)
	at com.google.devtools.build.lib.actions.Action.execute(Action.java:133)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$5.execute(SkyframeActionExecutor.java:957)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.continueAction(SkyframeActionExecutor.java:1124)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.run(SkyframeActionExecutor.java:1082)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.runStateMachine(ActionExecutionState.java:160)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.getResultOrDependOnFuture(ActionExecutionState.java:93)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.executeAction(SkyframeActionExecutor.java:516)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.checkCacheAndExecuteIfNeeded(ActionExecutionFunction.java:827)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.computeInternal(ActionExecutionFunction.java:323)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.compute(ActionExecutionFunction.java:161)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:571)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:382)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make java.lang.String(byte[],byte) accessible: module java.base does not "opens java.lang" to unnamed module @297e2f87
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Constructor.checkCanSetAccessible(Constructor.java:188)
	at java.base/java.lang.reflect.Constructor.setAccessible(Constructor.java:181)
	at com.google.devtools.build.lib.unsafe.StringUnsafe.<init>(StringUnsafe.java:61)
	at com.google.devtools.build.lib.unsafe.StringUnsafe.<clinit>(StringUnsafe.java:36)
	... 22 more

ERROR: Could not build Bazel
```

File: devel/bazel/Portfile (hunk 2 above)
Comment (essandess) [reply devel/bazel/Portfile:7]: The `bazel`-related ports require a complete refactoring, either with the approach used in this PR, or rewritten from scratch using `bazelisk`.

With justification, @cjones051073 threw in the towel on contributing to anything `bazel`-related because of the unreasonable resources necessary to address its breaking changes.

I am in this PR trying to streamline the port the achieve an up-to-date`bazel` version  that builds `tensorflow`, which, by the way, still doesn't build on macOS: https://github.com/tensorflow/tensorflow/issues/60191

I'd recommend viewing this effort as a full rewrite of the port using the existing approach.

File: _resources/port1.0/group/bazel-1.0.tcl (hunk 3 above)
Comment (essandess) [reply _resources/port1.0/group/bazel-1.0.tcl:282]: Nope.

File: devel/bazel/Portfile (hunk 4 above)
Comment (essandess) [reply devel/bazel/Portfile]: `revision` bumped.

**Comment by essandess** [comment]
> CI failed:
> 
> ```
> FATAL: bazel crashed due to an internal error. Printing stack trace:
> java.lang.ExceptionInInitializerError
> 	at com.google.devtools.build.lib.actions.ParameterFile.writeContentLatin1(ParameterFile.java:141)
>  …
> 	at com.google.devtools.build.lib.unsafe.StringUnsafe.<clinit>(StringUnsafe.java:36)
> 	... 22 more
> 
> ERROR: Could not build Bazel
> ```

I confirm that this builds and runs on:

macOS 13.3.1 22E261 x86_64
Xcode 14.3 14E222b

Good luck tracking down whatever java thing breaks on the CI 🙃

**Comment by catap** [comment]
@essandess which java version have you use?

**Comment by catap** [comment]
@essandess the root cause of issue: `java` PG doesn't respect `java.version 11`  and tries to build it on JDK17 and JDK17 is supported since 7.0.0: https://github.com/bazelbuild/bazel/pull/14040

**Comment by catap** [comment]
@essandess I've opened a PR into your branch which fixes CI: https://github.com/essandess/macports-ports/pull/2

Let unblock it!

**Comment by cjones051073** [comment]
> @essandess the root cause of issue: `java` PG doesn't respect `java.version 11` and tries to build it on JDK17 and JDK17 is supported since 7.0.0: [bazelbuild/bazel#14040](https://github.com/bazelbuild/bazel/pull/14040)
> 
> ```
> DEBUG: java-portgroup: Trying to find JVM version: 11
> DEBUG: java-portgroup: Detected JVMs: 8 /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.372-7/x64/Contents/Home/Users/runner/hostedtoolcache/Java_Adopt_jdk/8.0.372-7/x64/Contents/Home 17 /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.7-7/x64/Contents/Home 11 /Users/runner/hostedtoolcache/Java_Adopt_jdk/11.0.19-7/x64/Contents/Home/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
> DEBUG: java-portgroup: Discovered matching JAVA_HOME: /Users/runner/hostedtoolcache/Java_Adopt_jdk/11.0.19-7/x64/Contents/Home/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
> DEBUG: Discovered JAVA_HOME via /usr/libexec/java_home: /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.7-7/x64/Contents/Home
> ```

No, The java PG does its best to satistfy whatever requirements it is accessed to. In this case there is I think something fishy in the environment on the CI machine causing issues.

The message

```
java-portgroup: Discovered matching JAVA_HOME: /Users/runner/hostedtoolcache/Java_Adopt_jdk/11.0.19-7/x64/Contents/Home/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
```

Comes from 

https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/java-1.0.tcl#L90

and shows that an attempt to run `/usr/libexec/java_home -f -v 11` successfully returned that path.

However....

That path then appears to fail a test to confirm its a valid directory, at

https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/java-1.0.tcl#L105

as that is the only way the message

```
DEBUG: Discovered JAVA_HOME via /usr/libexec/java_home: /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.7-7/x64/Contents/Home
```

can appear, and at this point the java PG gives up trying to satisfy the requirements, as its run out of options to do so, and just uses the default, which in this case appears to be for version 17.

So the primary question is why does the path

```
/Users/runner/hostedtoolcache/Java_Adopt_jdk/11.0.19-7/x64/Contents/Home/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
```

get listed as a valid JM alternative, but then fail the `isdirectory` test.

I don't think there is anything wrong with the javaPG here, it is just doing its best to work with the enviroment it is presented with.

**Comment by catap** [comment]
@cjones051073 yes, the issue that Github runners has 3rd party JDK. As soon as I've removed it, the CI had passed.

I think that ignoring 3rd party JDKs isn't the right move. Cleaner to remove it from Github runners.

**Comment by cjones051073** [comment]
> @cjones051073 yes, the issue that Github runners has 3rd party JDK. As soon as I've removed it, the CI had passed.
> 
> I think that ignoring 3rd party JDKs isn't the right move. Cleaner to remove it from Github runners.

A priori there is nothing wrong with third party JDKs, assuming they function correctly... So for me the question I sk above is still relevant, why is the JDK that is made available failing the `isdirectory` test ?

**Comment by catap** [comment]
> > @cjones051073 yes, the issue that Github runners has 3rd party JDK. As soon as I've removed it, the CI had passed.
> > I think that ignoring 3rd party JDKs isn't the right move. Cleaner to remove it from Github runners.
> 
> A priori there is nothing wrong with third party JDKs, assuming they function correctly... So for me the question I ask above is still relevant, why is the JDK that is made available failing the `isdirectory` test ?

Because it isn't a directory :) More of that this path doesn't exists.

See, it returns a path:
```
java-portgroup: Discovered matching JAVA_HOME: /Users/runner/hostedtoolcache/Java_Adopt_jdk/11.0.19-7/x64/Contents/Home/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
```
and here `/Home/Users` appears twice. At the begining of the path, and in the moddle as `.../x64/Contents/Home/Users/runner/hostedtoolcache/...`

=> seems something is quite wrong with this JDK and and easy way is remove it, and use MacPorts JDK.

**Comment by cjones051073** [comment]
> > > > @cjones051073 yes, the issue that Github runners has 3rd party JDK. As soon as I've removed it, the CI had passed.
> > > > I think that ignoring 3rd party JDKs isn't the right move. Cleaner to remove it from Github runners.
> > > 
> > > 
> > > A priori there is nothing wrong with third party JDKs, assuming they function correctly... So for me the question I ask above is still relevant, why is the JDK that is made available failing the `isdirectory` test ?
> > 
> > 
> > Because it isn't a directory :) More of that this path doesn't exists.
> > See, it returns a path:
> > ```
> > java-portgroup: Discovered matching JAVA_HOME: /Users/runner/hostedtoolcache/Java_Adopt_jdk/11.0.19-7/x64/Contents/Home/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and here `/Home/Users` appears twice. At the begining of the path, and in the moddle as `.../x64/Contents/Home/Users/runner/hostedtoolcache/...`
> > => seems something is quite wrong with this JDK and and easy way is remove it, and use MacPorts JDK.
> 
> I do not agree with the changes you make in [essandess#2](https://github.com/essandess/macports-ports/pull/2) . They might well help, by removing the problematic path, but the proper fix is is for someone to look into _why_ the above path is being created and fix that instead. As I said, there is in principle nothing wrong with using a third party JDK as long as that JDK works.

So I guess my question now is who should the above buggy path be reported to ?

**Comment by catap** [comment]
Good, it doesn't fix it. Very good. I was too fast :)

**Comment by cjones051073** [comment]
> Nothing IMHO needs to be changed in the java PG here. It simply relies on the system returning a valid list of paths when it executes `/usr/libexec/java_home -V`. The issue here is that is including a buggy path. Thats not an issue with the java PG itself, but the underlying CI environment.

... or...

its possible what is being returned is being incorrectly munged by the proc in the java PG

https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/java-1.0.tcl#L171

However to judge if thats the case or not we need to see the *raw* output from 

```
/usr/libexec/java_home -V
```

when run on the CI machine.

**Comment by cjones051073** [comment]
> > However to judge if thats the case or not we need to see the _raw_ output from
> > ```
> > /usr/libexec/java_home -V
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > when run on the CI machine.
> 
> You're reading my mind and I'm running it right now. So, here the output:
> 
> ```
>   Matching Java Virtual Machines (5):
>   /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.7-7/x64/Contents/Home
>       17.0.7 (x86_64) "Eclipse Adoptium" - "OpenJDK 17.0.7" /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.7-7/x64/Contents/Home
>       11.0.19 (x86_64) "Eclipse Adoptium" - "OpenJDK 11.0.19" /Users/runner/hostedtoolcache/Java_Adopt_jdk/11.0.19-7/x64/Contents/Home
>       11.0.19 (x86_64) "Eclipse Adoptium" - "OpenJDK 11.0.19" /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
>       1.8.0_372 (x86_64) "Eclipse Temurin" - "Eclipse Temurin 8" /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.372-7/x64/Contents/Home
>       1.8.0_372 (x86_64) "Eclipse Temurin" - "Eclipse Temurin 8" /Users/runner/hostedtoolcache/Java_Adopt_jdk/8.0.372-7/x64/Contents/Home
> ```
> 
> => environment is innocent and issue from java PG, or at least seems so.

Yeah, it looks like the logic falls over when there are mulitple paths for exactly the same version. I guess this was a scenario never expected....

So the correct fix here is to attempt to deal with this in some way...

**Comment by cjones051073** [comment]
> > > However to judge if thats the case or not we need to see the _raw_ output from
> > > ```
> > > /usr/libexec/java_home -V
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > when run on the CI machine.
> > 
> > 
> > You're reading my mind and I'm running it right now. So, here the output:
> > ```
> >   Matching Java Virtual Machines (5):
> >   /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.7-7/x64/Contents/Home
> >       17.0.7 (x86_64) "Eclipse Adoptium" - "OpenJDK 17.0.7" /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.7-7/x64/Contents/Home
> >       11.0.19 (x86_64) "Eclipse Adoptium" - "OpenJDK 11.0.19" /Users/runner/hostedtoolcache/Java_Adopt_jdk/11.0.19-7/x64/Contents/Home
> >       11.0.19 (x86_64) "Eclipse Adoptium" - "OpenJDK 11.0.19" /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
> >       1.8.0_372 (x86_64) "Eclipse Temurin" - "Eclipse Temurin 8" /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.372-7/x64/Contents/Home
> >       1.8.0_372 (x86_64) "Eclipse Temurin" - "Eclipse Temurin 8" /Users/runner/hostedtoolcache/Java_Adopt_jdk/8.0.372-7/x64/Contents/Home
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > => environment is innocent and issue from java PG, or at least seems so.
> 
> Yeah, it looks like the logic falls over when there are mulitple paths for exactly the same version. I guess this was a scenario never expected....
> 
> So the correct fix here is to attempt to deal with this in some way...

The core issue here is the use of `dict append` when added new keys to the version dict

https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/java-1.0.tcl#L189

If you call that twice for exactly the same key (`$vers) then you get what we seem, the path string for that version is *all* paths munged together.

A simple fix is to use `dict set` instead. Then if called multiple times for the same key you get `last one wins`.

**Comment by cjones051073** [comment]
@catap Please can you try again with https://github.com/macports/macports-ports/pull/18496

**Comment by catap** [comment]
@cjones051073 thanks, you was faster than me :)

It is testing right now here: https://github.com/catap/macports-ports/actions/runs/4862643389

I really think that we had found the cause and it I feel that I've encountered it once but never had motivated enough to dig.

**Comment by cjones051073** [comment]
> @cjones051073 thanks, you was faster than me :)
> 
> It is testing right now here: https://github.com/catap/macports-ports/actions/runs/4862643389
> 
> I really think that we had found the cause and it I feel that I've encountered it once but never had motivated enough to dig.

@catap Those builds above are still using https://github.com/essandess/macports-ports/pull/2 as I see in the macOS13 logs messages you added there

```
2023-05-02T14:55:31.9724720Z ##[group]Uninstalling preinstalled JDKs
2023-05-02T14:55:31.9725400Z Moving directories...
2023-05-02T14:55:31.9907280Z /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk
```

Please cancel these and start them again without this MR...

**Comment by catap** [comment]
Sorry, tooo many branches!

The right one: https://github.com/catap/macports-ports/actions/runs/4862901100

**Comment by cjones051073** [comment]
> Sorry, tooo many branches!
> 
> The right one: https://github.com/catap/macports-ports/actions/runs/4862901100

OK, those seem to have worked. I see the warnings I would expect and a valid JVM path is now picked.

```
2023-05-02T15:24:10.4838430Z Warning: java-portgroup: Found multiple installations for JVM 11
2023-05-02T15:24:10.4839550Z Warning: java-portgroup: Found multiple installations for JVM 8
2023-05-02T15:24:10.4841620Z DEBUG: java-portgroup: Detected JVMs: 8 /Users/runner/hostedtoolcache/Java_Adopt_jdk/8.0.372-7/x64/Contents/Home 17 /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.7-7/x64/Contents/Home 11 /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
2023-05-02T15:24:10.4843350Z DEBUG: java-portgroup: Discovered matching JAVA_HOME: /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.19-7/x64/Contents/Home
```

I am reasonably happy https://github.com/macports/macports-ports/pull/18496 is an improvement over what is currently there, so I think I will merge it. One thing I do not do, as I wouldn't really know how, is make any decision on which path to use when there are multiple for the same (major) version. Currently its just whichever is the last listed by `java_home -V`.

**Comment by catap** [comment]
@cjones051073 seems that your PR works as expected

**Comment by catap** [comment]
@cjones051073 I've supprised that it doesn't require to be rebased

**Comment by cjones051073** [comment]
An, i assumed the tests would do that themselves...

**Comment by cjones051073** [comment]
@essandess Please rebase your branch here so we can retest.

**Comment by catap** [comment]
Yahoo!

**Comment by mascguy** [comment]
So are we ready to merge this then...?

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
Just tested on 10.15/Xcode 12, and I did encounter two issues:
* Building with ccache caused failures, so I added `configure.ccache no` locally. It might be something specific to my installation though, have you folks tested with ccache enabled?
* The "Xcode Locator" component - built via file `tools/osx/BUILD` - specifies two arches: both arm64, and x86_64. (Plus it tries to pass linker option `-no_adhoc_codesign`, which doesn't appear (?) to be valid for 10.15.) It was necessary to patch that file, via the following:

```
--- a/tools/osx/BUILD.orig	2023-05-03 17:34:35.000000000 -0400
+++ b/tools/osx/BUILD	2023-05-03 17:43:45.000000000 -0400
@@ -29,7 +29,7 @@
 
 DARWIN_XCODE_LOCATOR_COMPILE_COMMAND = """
   /usr/bin/xcrun --sdk macosx clang -mmacosx-version-min=10.13 -fobjc-arc -framework CoreServices \
-      -framework Foundation -arch arm64 -arch x86_64 -Wl,-no_adhoc_codesign -Wl,-no_uuid -o $@ $< && \
+      -framework Foundation -arch x86_64 -Wl,-no_uuid -o $@ $< && \
   env -i codesign --identifier $@ --force --sign - $@
 """
```

(The above patch isn't production-worthy, since we'd also want to insert the correct [single?] arch at patch time.)

With those two changes in play, though, `bazel` built successfully on 10.15. Great work you guys!

**Comment by essandess** [comment]
Thanks everyone for getting this working!

**Comment by mascguy** [comment]
Quick update: I'm now able to build Bazel for 10.14, by relaxing the Xcode/Clang minimum: Xcode 10.3/Clang 10 are sufficient, and the build looks good.

That change was included in Herby's update to Bazel 6.2, via PR: https://github.com/macports/macports-ports/pull/18601

As for `py-tensorflow`, builds fail for 10.15 and earlier, with error "argument list too long." Currently reviewing potential upstream fixes for that, which might involve patching Bazel. But making progress!
